src/libYARP_OS/src/RateThread.cpp

```diff
@@ -299,14 +320,45 @@ RateThread::~RateThread()
     }
 }
 
+bool RateThread::checkRequiredPeriod(int &period)
+{
+#ifndef YARP_NO_DEPRECATED // since YARP 2.3.72
+    if(period < 0)
+    {
+        yError() << "RateThread cannot have negative period.";
+        yAssert(false);  // condition already checked
+    }
+
+    if(period == 0)
+    {
+        // go ahead for now ...
+        yWarning() << "RateThread period cannot be 0 ms!\n" <<
+                      "  This setting may cause slowdowns in multithread environment, expecially when mutex are involved.\n" <<
+                      "  This usage is deprecated and it will not be allowed in future versions of YARP.\n" <<
+                      "  If no specific period is required, use yarp::os::Thread instead or set a valid period.";
+    }
+
+#else
+
+    if(period <= 0)
+    {
+        yError() << "RateThread period must be positive value.";
+        yAssert(false);  // condition already checked
+    }
+#endif
+    return true;
+}
+
 void RateThread::initWithSystemClock()
 {
     ((RateThreadCallbackAdapter*)implementation)->initWithSystemClock();
 }
 
 bool RateThread::setRate(int period)
 {
-    return ((RateThreadCallbackAdapter*)implementation)->setRate(period);
+    bool ret = checkRequiredPeriod(period);
```

Comment:
The function `checkRequiredPeriod()` either return `true` or stops the application with a `yAssert(false)`.
In my opinion this behaviour it is not clear.
My suggestion is to either change this behaviour, thus allowing a `return false`and then use a `yAssert()` on it or to not a return value at all.